test/integration/consul-container/metrics/leader_test.go

```diff
@@ -0,0 +1,113 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	libcluster "github.com/hashicorp/consul/integration/consul-container/libs/cluster"
+	"github.com/hashicorp/consul/integration/consul-container/libs/node"
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+	"github.com/hashicorp/consul/lib/retry"
+)
+
```

Comment:
Could you kindly add a small comment about what this test is meant to do? You could use the `given, when, then` framework if you'd like.

I think, especially for integ tests, this will help maintainability down the road.

Reply:
Done